src/libraries/System.Private.CoreLib/src/System/Globalization/Calendar.cs

```diff
@@ -711,7 +712,7 @@ internal static long TimeToTicks(int hour, int minute, int second, int milliseco
 
         internal static int GetSystemTwoDigitYearSetting(CalendarId CalID, int defaultYearValue)
         {
-            int twoDigitYearMax = GlobalizationMode.UseNls ? CalendarData.NlsGetTwoDigitYearMax(CalID) : CalendarData.IcuGetTwoDigitYearMax();
+            int twoDigitYearMax = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? CalendarData.NlsGetTwoDigitYearMax(CalID) : CalendarData.IcuGetTwoDigitYearMax();
```

Comment:
Thanks @cdbullard for adding that. We usually avoid doing the OS checks inside the code. Instead we separate the code during compile time. 

For this change, please move the method `NlsGetTwoDigitYearMax` to the file `CalendarData.Windows.cs` and rename the method to something like `GetTwoDigitYearMax`. Then move the method `IcuGetTwoDigitYearMax` to the file `CalendarData.Unix.cs` and rename the method to the same name `GetTwoDigitYearMax` (taking same parameters as the one moved to Windows file). last, make `GetSystemTwoDigitYearSetting` just `CalendarData.GetTwoDigitYearMax`.

Reply:
Thanks for clarifying that for me. If I make the method `GetSystemTwoDigitYearSetting` like this:

```
internal static int GetSystemTwoDigitYearSetting(CalendarId CalID, int defaultYearValue)
{
    return CalendarData.GetTwoDigitYearMax(CalID);
}
```

should I also remove the `defaultYearValue` from the signature and from the places this method is being called?

Reply:
No, please keep it as:

```C#
internal static int GetSystemTwoDigitYearSetting(CalendarId CalID, int defaultYearValue)
{
    int twoDigitYearMax = CalendarData.GetTwoDigitYearMax(CalID);
    return twoDigitYearMax >= 0 ? twoDigitYearMax : defaultYearValue;
}
```

Sorry I was not clear in that part :-)

Reply:
Thanks! I've just pushed a commit for this, let me know if any of it should be changed